### ISSUE TITLE
Can't reach database server when doing a large findMany query

### ISSUE THREAD

**Issue by hnrw** [prisma/prisma#9275]
### Bug description

When I try the following query, I get the following error:

```
const users = await p.user.findMany({
  include: { quiz: true },
})
```
```
(node:7642) UnhandledPromiseRejectionWarning: Error:
Invalid `prisma.user.findMany()` invocation:

 Can't reach database server at `localhost`:`5432`

Please make sure your database server is running at `localhost`:`5432`.
```

With some initial testing, I noticed removing the `include` argument allows the query to run just fine.
```
// works
const users = await p.user.findMany({}) 
```

Also, I could leave the `include` in if I limited the number of rows with `take`. Roughly 32,000 seemed to be the limit.

```
// works
const users = await p.user.findMany({
  include: { quiz: true },
  take: 32000,
})

// error
const users = await p.user.findMany({
  include: { quiz: true },
  take: 40000,
})
```

I think this is a bug. But if it's a known limitation, it would be good if there was a better error message. I initially spent a long time troubleshooting my connection to postgres.

EDIT: I just tested this with the app running on heroku with a heroku managed database. I get the exact same error, so it's not an issue with my local db.

### How to reproduce

Run:
```
p.user.findMany({
  include: { quiz: true },
})
```
Both `user` and `quiz` have about 100k rows.

### Prisma information
```
model User {
  id       String    @default(uuid())
  email    String    @id @unique
  customer Boolean   @default(false)
  name     String?
  acDate   DateTime?
  quiz     Quiz?
}

model Quiz {
  id         String    @id @default(uuid())
  firstName  String?
  skill      String
  playstyle  String
  discipline String
  age        String
  gender     String
  racketNum  Int?
  date       DateTime?
  user       User      @relation(fields: [userEmail], references: [email], onDelete: Cascade)
  userEmail  String
}
```

### Environment & setup

- OS:  Windows 10 running WSL2 Ubunutu 20.04
- Database: A local postgres instance
- Node.js version: v14.5.0

### Prisma Version

```
prisma                  : 3.0.2
@prisma/client          : 3.0.2
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 2452cc6313d52b8b9a96999ac0e974d0aedf88db
Studio                  : 0.423.0
```

**Comment by some-user123** [comment]
That sounds familiar ;-) #8832

**Comment by hnrw** [comment]
> That sounds familiar ;-) #8832

Interesting. If your case, does the query run correctly when limiting the number of rows with `take`?

**Comment by some-user123** [comment]
Yes, `take: 32766` works, `take: 32767` fails.

**Comment by nwidynski** [comment]
This is directly related to #8878

**Comment by NurbekGithub** [comment]
Had this problem today

**Comment by jameskennethrobinson** [comment]
Also having this problem on v3.3.0

**Comment by janpio** [comment]
Let's close this as a duplicate of https://github.com/prisma/prisma/issues/8832 then 👍

**Comment by jkomyno** [comment]
We have closed the original issue in `prisma@4.4.0`, please see [this comment](https://github.com/prisma/prisma/issues/8832#issuecomment-1260533913).
